### ISSUE TITLE
NPE in DocThumbTask

### ISSUE THREAD

**Issue by fmpfeifer** [sepinf-inc/IPED#872]
Processing aborted with NPE while processing doc file. This is the log section (actual file name redacted):

```
2021-12-12 13:27:24	[ERROR]	[gpinf.indexer.IndexFiles]			Processing Error: 
java.lang.Exception: Worker-34 Error while processing /MAT672/vol_vol8/pagefile.sys>>Carved-6989516800.zip>>XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.doc (nullbytes)
	at dpf.sp.gpinf.indexer.process.Worker.process(Worker.java:181) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.Worker.processNewItem(Worker.java:215) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.ParsingTask.parseEmbedded(ParsingTask.java:652) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.parsers.PackageParser.parseEntry(PackageParser.java:431) ~[iped-parsers-impl-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.parsers.PackageParser.handleEntry(PackageParser.java:237) ~[iped-parsers-impl-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.parsers.PackageParser.parse(PackageParser.java:196) ~[iped-parsers-impl-3.18.11.jar:?]
	at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:280) ~[tika-core-1.20.jar:1.20]
	at dpf.sp.gpinf.indexer.parsers.IndexerDefaultParser.parse(IndexerDefaultParser.java:233) ~[iped-parsers-impl-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.io.ParsingReader$BackgroundParsing.run(ParsingReader.java:251) ~[iped-engine-3.18.11.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_152]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_152]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:1.8.0_152]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:1.8.0_152]
	at java.lang.Thread.run(Unknown Source) ~[?:1.8.0_152]
Caused by: java.lang.NullPointerException
	at dpf.sp.gpinf.indexer.process.task.ThumbTask.hasThumb(ThumbTask.java:37) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.DocThumbTask.process(DocThumbTask.java:229) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processMonitorTimeout(AbstractTask.java:221) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:159) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:201) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:167) ~[iped-engine-3.18.11.jar:?]
	at dpf.sp.gpinf.indexer.process.Worker.process(Worker.java:172) ~[iped-engine-3.18.11.jar:?]
	... 13 more
```
IPED version: 3.18.11

after disabling DocThumbs and restarting with "--continue" I was able to get the processing going.
I can provide the offending file for testing in PV.

Thanks

**Comment by tc-wleite** [comment]
Hi @fmpfeifer! 
Can you please send me the triggering file?

**Comment by fmpfeifer** [comment]
Sure. I'm sending via Teams.
Thanks

**Comment by tc-wleite** [comment]
Thanks!
I am out for a few days, but will look into this as soon as I return.

**Comment by lfcnassif** [comment]
Thanks @fmpfeifer and @tc-wleite. I think this is related to #514, I possibly forgot to cherry pick its solution to 3.18.x when cherry picking the whole feature from master. @fmpfeifer if you can test a possible fix, I can push to some branch.

**Comment by fmpfeifer** [comment]
Hi @lfcnassif, sure, I can test it. Thanks

**Comment by lfcnassif** [comment]
just pushed to #872_fix branch.

**Comment by lfcnassif** [comment]
> If you want to test yourself, I can send you a 17KB file that triggers the issue.

Sure.

**Comment by fmpfeifer** [comment]
Sent via Teams. Thanks.

**Comment by lfcnassif** [comment]
fixed by 96e6156
